### DISCUSSION TITLE
Notification when a specific person is detected

### DISCUSSION THREAD

**Discussion by Sceptersax** [blakeblackshear/frigate#7492] [General]
I'm running compreface as my detector and I'm using double-take as the UI to train but I'd like to know how to send a notification when a specific person is detected. 

I'm really new to this and have very little experience with API's but as I understand it, I can use frigates ability to detect a person, an API is used between frigate and compreface to to trigger facial recognition. 

I also understand that I can use an API to send a recent detection (latest) wherever (let's say home assistant), but how would I use an API to only send a specific trained person (or unknown) to HA?

Thanks in advance.

**Comment by NickM-27** [comment]
In HA you can check the `sub_label` field of the payload in /events, that will contain the name doubletake has sent OR null if it hasn't been set.

**Reply by Sceptersax** [reply]
Is this the same for "unknown"?

**Reply by NickM-27** [reply]
No, it will not but set to unknown. If a person has been detected but the sub label is null that means no match has been detected.

**Reply by Sceptersax** [reply]
Got it. Thanks @NickM-27

**Reply by Sceptersax** [reply]
I also found this in the double-take github page:


```

MQTT

Publish results to double-take/matches/<name> and double-take/cameras/<camera>. The number of results will also be published to double-take/cameras/<camera>/person and will reset back to 0 after 30 seconds.

Errors from the API will be published to double-take/errors.

mqtt:
  host: localhost

double-take/matches/david

{
  "id": "1623906078.684285-5l9hw6",
  "duration": 1.26,
  "timestamp": "2021-06-17T05:01:36.030Z",
  "attempts": 3,
  "camera": "living-room",
  "zones": [],
  "match": {
    "name": "david",
    "confidence": 66.07,
    "match": true,
    "box": { "top": 308, "left": 1018, "width": 164, "height": 177 },
    "type": "latest",
    "duration": 0.28,
    "detector": "compreface",
    "filename": "2f07d1ad-9252-43fd-9233-2786a36a15a9.jpg",
    "base64": null
  }
}

```
So I guess you can use the "double-take/cameras/<camera>/" event to specify the person's name.

**Reply by NickM-27** [reply]
yes, but I don't see a reason to use 2 separate mqtt topics when a single one has all the necessary information.

**Reply by Sceptersax** [reply]
Well, I also have another endpoint I might consider other than HA. I figure, I should probably start with the approach that can suffice both.
